### PR TITLE
Add prettier. Closes #466

### LINKED ISSUE #466: Code formatting
PR: https://github.com/datproject/dat-desktop/pull/481

### PR DESCRIPTION
**Pull request by juliangruber**
based upon https://github.com/datproject/dat-desktop/pull/480, this will help us even more with consistency across all contributors.

Some of prettier's rules can be tweaked as well, but for now I value consistency a lot over the "most optimal way" of writing a line of code.

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -98,29 +103,44 @@ tap('working with dats', function (t) {
...
+    )
     .then(() => endTest(app))
 })
 
 // Create a new app instance
 function createApp () {
   return new spectron.Application({
     path: path.join(__dirname, '../node_modules/.bin/electron'),
-    args: [path.join(__dirname, '../index.js'), '--data', TEST_DATA, '--db', TEST_DATA_DB],
+    args: [
```
Comment (martinheidegger) [inline tests/index.js]: Is it just me or has this become significantly less readable.

File: tests/index.js [hunk 2]
```diff
@@ -12,83 +12,88 @@ tap('init', function (t) {
     var app = createApp()
     return waitForLoad(app)
       .then(() => app.browserWindow.isVisible())
-      .then((val) => t.ok(val, 'isVisible'))
+      .then(val => t.ok(val, 'isVisible'))
```
Comment (martinheidegger) [inline tests/index.js]: Just noticed that the #453 needs to be merge into this.

File: tests/index.js [hunk 3]
```diff
@@ -98,29 +103,44 @@ tap('working with dats', function (t) {
...
 function waitForLoad (app, t) {
-  return app.start().then(function () {
-    return app.client.waitUntilWindowLoaded()
-  }).then(function () {
-    // Switch to the main window
-    return app.client.windowByIndex(0)
-  }).then(function () {
-    return app.client.waitUntilWindowLoaded()
-  })
+  return app
```
Comment (martinheidegger) [inline tests/index.js]: This also seems way more stringent in the former version.

File: tests/index.js (hunk 1 above)
Comment (juliangruber) [reply tests/index.js]: yes, it's not perfect every single time, but it's consistent

Comment (juliangruber) [reply tests/index.js]: before, this line was 91 chars long, limiting lines to 80 chars is a common practice

**Comment by juliangruber** [comment]
I'm going to make a move on this now because I don't want to have to deal with too many merge conflicts and value having a formatter _a lot_. The success of [prettier](https://www.npmjs.com/package/prettier) and languages like go that have this built on speaks about the advantages.

Not having to argue about the best way of writing something ever again, in a collaborative environment, is usually worth more than that one _perfect_ way of writing a code block.

**Pull request merged**